src/App.tsx

```diff
@@ -1,24 +1,222 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import {
+  USER_ID,
+  createTodo,
+  destroyTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoInfo } from './components/TodoInfo/TodoInfo';
+import { ErrorType } from './types/ErrorTypes';
+import { UpdateDataTodo } from './types/UpdateDataTodo';
 
-const USER_ID = 0;
+type IsCompleted = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedFilter, setCompletedFilter] = useState<IsCompleted>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUploadingTitle, setIsUploadingTitle] = useState(false);
+  const [isLoadingTodoIds, setIsLoadingTodoIds] = useState<number[]>([]);
+  const [isAllCompleted, setIsAllCompleted] = useState(false);
+  const [errors, setErrors] = useState<ErrorType[]>([]);
+
+  const addError = (error: ErrorType) => {
+    setErrors(currentErrors => [error, ...currentErrors]);
+    setTimeout(() => {
```

Comment:
move this `setTimeout` to the `Error` component, bc you need to clear your timeout